src/test/ui/nll/closure-requirements/escape-argument-callee.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Test closure that:
+//
+// - takes an argument `y`
+// - stores `y` into another, longer-lived spot
+//
+// *but* the signature of the closure doesn't indicate that `y` lives
```

Comment:
but in the closure's signature, `y` and the slot are different unrelated higher-ranked regions. Because closure inference is not allowed to add where-clauses between higher-ranked regions, there is no set of free lifetimes that allows the closure to run, so the closure reports the error (and hence we see it before the closure's "external requirements" report).